packages/button/src/index.tsx

```diff
@@ -21,7 +22,42 @@ type StyledComponentElement =
   | keyof JSX.IntrinsicElements
   | React.ComponentType<any>
 
-const StyledText = styled(Text)<StyledTextProps>`
+interface ButtonIconAttachmentProps {
+  iconPosition?: string
```

Comment:
Change `iconPosition` to enum or a constant typed set so it would be harder to make a mistake when using it.
Update: just like below in `styled-components` augmentation declaration, you can put `'left' | 'center'` instead of `string` here

Reply:
done